app/components/Notes.tsx

```diff
@@ -0,0 +1,57 @@
+"use client";
+import React, { useEffect, useState } from "react";
+import { fetchNotes } from "./actions";
+import Link from "next/link";
+
+interface Notes {
+  _id: string;
+  title: string;
+  content: string;
+  category: string;
+  admin: string;
+}
+const Notes = () => {
+  const [notes, setNotes] = useState<Notes[]>([]);
+  const fetchData = async () => {
+    const data = await fetchNotes();
+    setNotes(data);
+  };
+  useEffect(() => {
+    fetchData();
+  }, []);
+  return (
+    <div>
+      {notes?.length > 0 ? (
+        <ul className="flex flex-col gap-2">
+          {notes!.map((note) => (
```

Comment:
this is not a good approach use this instead
```suggestion
      {notes && notes.length > 0 ? (
        <ul className="flex flex-col gap-2">
          {notes.map((note) => (
```